Build error with gold linker on Power9 little-endian

**Intention**
Attempted to build current head.

**Describe the bug**
Building the RTlib on Power9 triggers an internal linker error in gold:

```
[ 53%] Linking C shared library libdyninstAPI_RT.so
/software/ml/CMake/3.13.3-GCCcore-8.2.0/bin/cmake -E cmake_link_script CMakeFiles/dyninstAPI_RT.dir/link.txt --verbose=1
/sw/installed/GCCcore/8.2.0/bin/cc -fPIC  -m64  -fvisibility=hidden -W -Wall -Wpointer-arith -Wcast-qual -Wcast-align -m64  -O2 -g   -Wl,--no-undefined  -shared -Wl,-soname,libdyninstAPI_RT.so.10.1 -o libdyninstAPI_RT.so.10.1.0 CMakeFiles/dyninstAPI_RT.dir/src/RTcommon.c.o CMakeFiles/dyninstAPI_RT.dir/src/RTmemEmulator.c.o CMakeFiles/dyninstAPI_RT.dir/src/RTposix.c.o CMakeFiles/dyninstAPI_RT.dir/src/RTlinux.c.o CMakeFiles/dyninstAPI_RT.dir/src/RTheap.c.o CMakeFiles/dyninstAPI_RT.dir/src/RTheap-linux.c.o CMakeFiles/dyninstAPI_RT.dir/src/RTthread.c.o CMakeFiles/dyninstAPI_RT.dir/src/RTspace.S.o CMakeFiles/dyninstAPI_RT.dir/src/RTsignal.c.o CMakeFiles/dyninstAPI_RT.dir/src/RTstatic_ctors_dtors_begin.c.o CMakeFiles/dyninstAPI_RT.dir/src/RTstatic_ctors_dtors_end.c.o CMakeFiles/dyninstAPI_RT.dir/src/RTthread-powerpc.c.o CMakeFiles/dyninstAPI_RT.dir/src/RTthread-powerpc-asm.S.o CMakeFiles/dyninstAPI_RT.dir/src/RTstatic_ctors_dtors-ppc64.c.o -ldl 
/sw/installed/binutils/2.31.1-GCCcore-8.2.0/bin/ld.gold: internal error in get_output_view, at output.h:111
collect2: error: ld returned 1 exit status
```

`get_output_view` is trying to ensure that everything being linked has sane boundaries:

```
 105   // Get a buffer to use to write to the file, given the offset into
 106   // the file and the size.
 107   unsigned char*
 108   get_output_view(off_t start, size_t size)
 109   {
 110     gold_assert(start >= 0
 111                 && start + static_cast<off_t>(size) <= this->file_size_);
 112     return this->base_ + start;
 113   }
```
So something is somehow running off the end of the file according to `gold`--or it has a negative start, but that's much less likely than that our .S files have done something weird.

**To Reproduce**
On the Power9 partition here at TUD:
```
git clone https://github.com/dyninst/dyninst
module load GCC Boost CMake M4 // gcc 8.2.x toolchain, boost 1.70, CMake 3.13
mkdir ppc64build
cd ppc64build
cmake ../dyninst -DCMAKE_INSTALL_PREFIX=/some/install/dir // note: install prefix apparently must be writable even when not performing a make install these days? By design?
make // builds with downloaded TBB, elfutils
```
**Expected behavior**
Successful build

**System:**
Power 9 little-endian, Linux
GCC 8.2.x toolchain
Commit d8a5fb05bf9a255482686078306dd7980249053d

--- a/cmake/optimization.cmake
+++ b/cmake/optimization.cmake
@@ -4,7 +4,7 @@ if (CMAKE_COMPILER_IS_GNUCXX
     OR ${CMAKE_C_COMPILER_ID} MATCHES Intel)
 if(ENABLE_LTO)
     set(LTO_FLAGS "-flto")
-    set(LTO_LINK_FLAGS "-fuse-ld=gold")
+    set(LTO_LINK_FLAGS "-fuse-ld=bfd")
 else()
     set(LTO_FLAGS "")
     set(LTO_LINK_FLAGS "")

